crates/server/src/store/in_memory/authority.rs

```diff
@@ -1296,6 +1456,148 @@ impl Authority for InMemoryAuthority {
     ) -> Result<Self::Lookup, LookupError> {
         Ok(AuthLookup::default())
     }
+
+    #[cfg(feature = "dnssec")]
+    async fn get_nsec3_records(
+        &self,
+        name: &LowerName,
+        query_type: RecordType,
+        lookup_options: LookupOptions,
+    ) -> Result<Self::Lookup, LookupError> {
+        // Function to compute the hash of a given name.
+        let hash_name = |name: &Name| {
+            let Nsec3Config {
+                hash_algorithm,
+                ref salt,
+                iterations,
+                ..
+            } = self.nsec3_config;
+            hash_algorithm.hash(salt, name, iterations).unwrap()
```

Comment:
There's a bug in the proto crate that masks this (I'll submit a separate PR for this,) but with that fixed, it is definitely possible to make this panic on a long zone name.